scripts/build_bus_regions.py

```diff
@@ -293,4 +297,11 @@ def get_id(coords):
         # remove empty polygons
         offshore_regions = offshore_regions[~offshore_regions.geometry.is_empty]
 
-    save_to_geojson(offshore_regions, snakemake.output.regions_offshore)
+    # save_to_geojson(offshore_regions, snakemake.output.regions_offshore)
+    # pd.concat(onshore_regions, ignore_index=True).to_file(snakemake.output.regions_onshore)
+    if offshore_regions:
+        pd.concat(offshore_regions, ignore_index=True).to_file(
+            snakemake.output.regions_offshore
+        )
+    else:
+        offshore_shapes.to_frame().to_file(snakemake.output.regions_offshore)
```

Comment:
Could we try to reduce this? ( I know this is not reduced in PyPSA-Eur)

 I think the first IF applies if there is a shape. The ELSE applied if there is not shape there.

Looking at [Stackoverflow ](https://stackoverflow.com/questions/70214336/can-i-avoid-a-valueerror-concatenating-empty-dataframes) and their last comment. Could we try something like:

```
        pd.concat([offshore_regions, pd.Dataframe()], ignore_index=True).to_file(
            snakemake.output.regions_offshore
        )
```

Instead of line 302-307?

Reply:
better than one IF is no IF :)